### DISCUSSION TITLE
C# Glue in sync with GDScript?

### DISCUSSION THREAD

**Discussion by a2937** [godotengine/godot-proposals#11230] [Scripting]
Hi all. I'm a little confused about something so I hope you don't mind. I see a handful of methods that aren't implemented in the C# API without any applicable built-in replacements like `string.reverse` and `len` among others. Are they something that the maintainers are open to accepting implementations for?

**Comment by theraot** [comment]
In general, yes.

You would find or open an issue, fork the repository, do the changes in the fork (creating a branch is recommended), then make a pull requests stating that it closes the issue. You can see the workflow in detail in the official documentation.

Once the pull request is open, it would be reviewed, and any necesary changes would pointed to you (e.g. missing documentation or test cases).

But know the review might take a while. As the team is often busy, so I advice you to have patience. It might also happen that there is some other pull request that the team prefers or the team have other plans.

In this case in particular, I do not know what is the approach the team has around `string` on C#. *Given that .NET has its own `String` type, I guess the approach is to make extension methods, but that is just my guess. So, I'd say open an issue and wait for comments.*
